content/en/blog/_posts/2022-11-14-forensic-container-checkpointing/index.md

```diff
@@ -0,0 +1,207 @@
+---
+layout: blog
+title: "Forensic container checkpointing in Kubernetes"
+date: 2022-11-14
+slug: forensic-container-checkpointing-alpha
+---
+
+**Authors:** Adrian Reber (Red Hat)
+
+Forensic container checkpointing allows the creation of stateful copies of a
+running container without the container knowing that it is being checkpointed.
+The copy of the container can be analyzed and restored in a sandbox environment
+multiple times without the original container being aware of it. Forensic
+container checkpointing was introduced as an alpha feature in Kubernetes v1.25.
+
+## How does it work?
+
+Forensic container checkpointing is based on
+[Checkpoint/Restore In Userspace][criu] (CRIU). With the
+help of CRIU it is possible to checkpoint and restore containers.
+CRIU is integrated in runc, crun, CRI-O and containerd and
+forensic container checkpointing as implemented in Kubernetes
+uses these existing CRIU integrations.
+
+[criu]: https://criu.org/
+
+## Why is it important?
+
+With the help of CRIU and the corresponding integrations it is possible to get
+all information and state about a running container on disk for later forensic
+analysis. Forensic analysis might be important to inspect a suspicious
+container without stopping or influencing it. If the container is really under
+attack, the attacker might detect attempts to inspect the container. Taking a
+checkpoint and analysing the container in a sandboxed environment offers the
+possibility to inspect the container without the original container and maybe
+attacker being aware of the inspection.
+
+In addition to the forensic container checkpointing use case, it is also
+possible to migrate a container from one node to another node without loosing
+the internal state. Especially for stateful containers with long initialization
+times restoring from a checkpoint might save time after a reboot or enable much
+faster startup times.
+
+## How do I use container checkpointing?
+
+The feature is behind a [feature gate][container-checkpoint-feature-gate], so
+make sure to enable the `ContainerCheckpoint` gate before you can use the new
+feature.
+
+The runtime must also support container checkpointing:
+
+* containerd: support is currently under discussion. See containerd
+  pull request [#6965][containerd-checkpoint-restore-pr] for more details.
+
+* CRI-O: v1.25 has support for forensic container checkpointing.
+
+[containerd-checkpoint-restore-pr]: https://github.com/containerd/containerd/pull/6965
+[container-checkpoint-feature-gate]: https://kubernetes.io/docs/reference/command-line-tools-reference/feature-gates/
+
+### Usage example with CRI-O
+
+To use forensic container checkpointing in combination with CRI-O, the runtime needs
+to be started with the command-line options `--enable-criu-support=true` and
+`--drop-infra-ctr=false`. For Kubernetes, you need to run your cluster with the
```

Comment:
We have spoken outside of here about my issues with this flag being here. It would be my preference to wait on this blog post to have support for `drop-infra-ctr=true` in main and 1.25, and then we can drop this stipulation . WDYT about that?

Reply:
I appreciate it :smile: if there ends up being some technical limitation or issue, I am not stubborn about it. But I'd love it if the blog went out without that line